Clever/wag
# add composite fields

see https://clever.atlassian.net/browse/INFRA-3311 for a detailed description

My suggested path for how to review this:
- start with the new models in the example swagger spec: https://github.com/Clever/wag/compare/INFRA-3311-add-composite-fields?expand=1#diff-45ad9b74c3d577f8f93330a424152cc9 there are two models that have a new `CompositeAttributes` config
- See how it gets autogenerated into
1. The db interface: https://github.com/Clever/wag/compare/INFRA-3311-add-composite-fields?expand=1#diff-4da4362dff6935a57c1dbbee1542176c
2. The implementation of these interface methods: https://github.com/Clever/wag/compare/INFRA-3311-add-composite-fields?expand=1#diff-d12c64d58f1e74735d893fc6c43eea36 and https://github.com/Clever/wag/compare/INFRA-3311-add-composite-fields?expand=1#diff-7e3dce76c75993f510e6682bd62719b4
3. Tests of the implementation: https://github.com/Clever/wag/compare/INFRA-3311-add-composite-fields?expand=1#diff-7ffa51045ea51f98311d849ab9fecb73

... the rest is only worth reviewing if you really want to dig in to how the sausage is made (i.e. tmpl files, gendb.go, templatefuncs.go)
